code/framework/api/src/main/java/io/cattle/platform/api/auth/Identity.java

```diff
@@ -0,0 +1,97 @@
+package io.cattle.platform.api.auth;
+
+import io.github.ibuildthecloud.gdapi.annotation.Field;
+import io.github.ibuildthecloud.gdapi.annotation.Type;
+
+@Type(name = "identity", pluralName = "identities")
+public class Identity {
+
+    private final String externalId;
+    private final String profilePicture;
+    private final String name;
+    private final String kind;
+    private final String profileUrl;
+
+    @Field(required = false, nullable = true)
+    public String getName() {
+        return name;
+    }
+
+    @Field(required = true, nullable = false)
+    public String getExternalId() {
+        return externalId;
+    }
+
+    @Field(required = true, nullable = false)
+    public String getKind() {
+        return kind;
+    }
+
+    @Field(required = false, nullable = true)
+    public String getId() {
+        return kind + ':' + externalId;
+    }
+
+    @Field(required = false, nullable = true)
+    public String getProfilePicture() {
+        return profilePicture;
+    }
+
+    @Field(required = false, nullable = true)
+    public String getProfileUrl() {
+        return profileUrl;
+    }
+
+    public Identity(String identityType, String externalId) {
+        this(identityType, externalId, null);
+    }
+
+    public Identity(String identityType, String externalId, String name) {
+        this(identityType, externalId, name, null, null);
+    }
+//
+//    public Identity(String identityType, String externalId, String name, String profileUrl) {
+//        this(identityType, externalId, name, profileUrl, null);
+//    }
+
+    public Identity(String identityType, String externalId, String name, String profileUrl, String profilePicture) {
+        this.externalId = externalId;
+        this.name = name;
+        this.kind = identityType;
+        this.profileUrl = profileUrl;
+        if (profilePicture == null) {
+            this.profilePicture = "http://robohash.org/" + getId() + ".png?set=set2";
+        } else {
+            this.profilePicture = profilePicture;
+        }
+    }
+
+    @Override
+    public String toString() {
+        return "Identity{" +
+                "id='" + getId() +'\'' +
+                "externalId=" + externalId + '\'' +
+                ", profilePicture='" + profilePicture + '\'' +
+                ", name='" + name + '\'' +
+                ", kind='" + getKind() + '\'' +
+                '}';
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```

Comment:
Your hash code and equals methods are based on different fields, is this intentional?  You might look at use HashCodeBuilder and ToStringBuilder from lang3 for this